RoddyCore/src/de/dkfz/roddy/core/RuntimeService.groovy

```diff
@@ -444,119 +446,143 @@ public class RuntimeService {
+        if (outputBaseDirectory.toString() == "")
+            throw new ConfigurationError("outputBaseDirectory not set. Consider --useiodir command-line option.",
+                    configuration, ConfigurationConstants.CFG_OUTPUT_BASE_DIRECTORY, null)
+        return outputBaseDirectory
     }
 
-    public File getInputFolderForAnalysis(Analysis analysis) {
-        return analysis.getConfiguration().getConfigurationValues().get(ConfigurationConstants.CFG_INPUT_BASE_DIRECTORY).toFile(analysis);
+    File getInputFolder(DataSet dataSet, Analysis analysis) {
+        File analysisInFolder = new File(getInputFolder(analysis).absolutePath + FileSystemAccessProvider.instance.getPathSeparator() + dataSet.getId())
+        return analysisInFolder
     }
 
-    public File getOutputFolderForProject(ExecutionContext context) {
-        return context.getConfiguration().getConfigurationValues().get(ConfigurationConstants.CFG_OUTPUT_BASE_DIRECTORY).toFile(context);
+    File getInputFolder(Analysis analysis) {
+        return getInputBaseDirectory(analysis.configuration).toFile(analysis)
     }
 
-    public File getOutputFolderForAnalysis(Analysis analysis) {
-        return analysis.getConfiguration().getConfigurationValues().get(ConfigurationConstants.CFG_OUTPUT_BASE_DIRECTORY).toFile(analysis);
+    File getOutputFolder(ExecutionContext context) {
+        return getOutputBaseDirectory(context.configuration).toFile(context)
     }
 
-    public File getInputFolderForDataSetAndAnalysis(DataSet dataSet, Analysis analysis) {
-        File analysisInFolder = new File(getInputFolderForAnalysis(analysis).absolutePath + FileSystemAccessProvider.instance.getPathSeparator() + dataSet.getId());
-        return analysisInFolder;
+    File getOutputFolder(Analysis analysis) {
+        return getOutputBaseDirectory(analysis.configuration).toFile(analysis)
+    }
+
+    /** Get the value of the outputAnalysisBaseDirectory variable given the ExecutionContext. If that variable is not defined fall back to
+     *  outputBaseDirectory. In both cases context variables (e.g. "${dataset}") are substituted if possible.
+     *
+     * @param context
+     * @return
+     */
+    File getOutputAnalysisFolder(ExecutionContext context) {
+        def outputBaseDirectory = getOutputFolder(context.analysis)
+        return context.getConfiguration().getConfigurationValues().
+                get(ConfigurationConstants.CFG_OUTPUT_ANALYSIS_BASE_DIRECTORY, outputBaseDirectory.toString()).toFile(context)
     }
 
-    public File getOutputFolderForDataSetAndAnalysis(DataSet dataSet, Analysis analysis) {
-        getOutputFolderForAnalysis(analysis);
-        File analysisOutFolder = analysis.getConfiguration().getConfigurationValues().get(ConfigurationConstants.CFG_OUTPUT_ANALYSIS_BASE_DIRECTORY).toFile(analysis, dataSet);
-        return analysisOutFolder;
+    File getOutputAnalysisFolder(DataSet dataSet, Analysis analysis) {
```

Comment:
Did you write or extend tests for the changed folder methods?

Reply:
Nope. I am somewhat hesitant to write more and more integration tests (everything accessing the filesystem is an integration test). But you are right that there should be some tests.